File: go/leap_year.go
package main

import "fmt"

func main() {
	var year int

	fmt.Printf("Type a number: ")
	fmt.Scan(&year)

	if year%4 == 0 {
		if year%100 != 0 && year%400 != 0 {
			fmt.Println("Leap year.")
		} else if year%100 == 0 {
			fmt.Println("Not leap year.")
		} else {
			fmt.Println("Leap year.")
		}
	} else {
		fmt.Println("Not leap year.")
	}
}

File: rust/leap_year.rs
use std::io::stdin;

fn main() {
    println!("Which year do you want to check? :");
    let mut year_input = String::new();
    stdin().read_line(&mut year_input).unwrap();

    let year: u32 = year_input.trim().parse().unwrap();

    if year % 4 == 0 {
        if year % 100 != 0 && year % 400 != 0 {
            println!("Leap year.");
        } else if year % 100 == 0 {
            println!("Not leap year");
        } else {
            println!("Leap year.");
        }
    } else {
        println!("Not leap year.")
    }
}
